packages/status-js/src/utils/deserialize-public-key.ts

```diff
@@ -0,0 +1,73 @@
+import { Point } from 'ethereum-cryptography/secp256k1'
+import {
+  toHex,
+  // utf8ToBytes as toBytes, // see https://github.com/paulmillr/noble-hashes/blob/d76eb7c818931d290c4c27abb778e8e269895154/src/utils.ts#L91-L96
+} from 'ethereum-cryptography/utils'
+import { varint } from 'multiformats'
+import { base58btc } from 'multiformats/bases/base58'
+
+/**
+ * @see https://github.com/multiformats/multibase/blob/af2d36bdfaeaca453d20b18542ca57bd56b51f6c/README.md#multibase-table
+ */
+const VALID_MULTIBASE_CODES = [
+  'f', // hexadecimal
+  'z', // base58btc
+] as const
+
+type MultibaseCode = typeof VALID_MULTIBASE_CODES[number]
+
+/**
+ * @see https://pkg.go.dev/github.com/multiformats/go-multicodec#pkg-types
+ */
+const VALID_MULTICODEC_CODES = [
+  231, // secp256k1-pub (compressed) (0xe7)
+] as const
+
+type MulticodecCode = typeof VALID_MULTICODEC_CODES[number]
+
+/**
+ * @see https://specs.status.im/spec/2#public-key-serialization for specification
+ */
+export function deserializePublicKey(
+  publicKey: string, // uncompressed, compressed, or compressed & encoded
+  options = { compress: true }
+): string {
+  const cpk = publicKey.replace(/^0[xX]/, 'f') // ensure multibase code for hexadecimal encoding
```

Comment:
Uses these shortcuts (i.e. cpk, c, v, p) for names, because I honestly couldn't come up with anything better and making more sense then just considering them in their context.

Reply:
Maybe at least `compressed` for line 35 and `code` for line 36?

Reply:
This ensures multibase code (`f`) for hexadecimal encoding, so it'd rather be something like `publicKeyWithMultibaseCode`.

Reply:
Addressed in https://github.com/status-im/status-web/pull/341/commits/e2ff431133e0ccc2a7581e2a0319cd5f92c1662c